test/test/unit/test_link.py

```diff
@@ -265,153 +276,37 @@ def test_remove_outlet(self, outlet1, outlet2):
-
-        inlet1.try_start.assert_called()
-        outlet1.try_start.assert_called()
-        outlet2.try_start.assert_not_called()
-
-    @patch(fqname(Outlet), spec=Outlet)
-    @patch(fqname(Outlet), spec=Outlet)
-    @patch(fqname(Inlet), spec=Inlet)
-    def test_on_start_outlet_exception_catch(self, inlet1, outlet1, outlet2):
-        logging.getLogger('databay.Link').setLevel(logging.WARNING)
-        outlet1.try_start.side_effect = lambda: exec('raise(RuntimeError())')
-        link = Link([inlet1], [outlet1, outlet2], timedelta(seconds=1), name='test_on_start', catch_exceptions=True)
-
-        with self.assertLogs(logging.getLogger('databay.Link'), level='ERROR') as cm:
-            link.on_start()
-        self.assertTrue('on_start outlet exception: "" for outlet:' in ';'.join(cm.output), cm.output)
-
-        inlet1.try_start.assert_called()
-        outlet1.try_start.assert_called()
-        outlet2.try_start.assert_called()
-
-    @patch(fqname(Outlet), spec=Outlet)
-    @patch(fqname(Inlet), spec=Inlet)
-    def test_on_shutdown_inlet_exception_raise(self, inlet1, outlet1):
-        inlet1.try_shutdown.side_effect = lambda: exec('raise(RuntimeError())')
-        link = Link([inlet1], [outlet1], timedelta(seconds=1), name='test_on_shutdown')
-
-        self.assertRaises(RuntimeError, link.on_shutdown)
-
-        inlet1.try_shutdown.assert_called()
-        outlet1.try_shutdown.assert_not_called()
-
-    @patch(fqname(Outlet), spec=Outlet)
-    @patch(fqname(Inlet), spec=Inlet)
-    def test_on_shutdown_inlet_exception_catch(self, inlet1, outlet1):
-        logging.getLogger('databay.Link').setLevel(logging.WARNING)
-        inlet1.try_shutdown.side_effect = lambda: exec('raise(RuntimeError())')
-        link = Link([inlet1], [outlet1], timedelta(seconds=1), name='test_on_shutdown', catch_exceptions=True)
-
-        with self.assertLogs(logging.getLogger('databay.Link'), level='ERROR') as cm:
-            link.on_shutdown()
-        self.assertTrue('on_shutdown inlet exception: "" for inlet:' in ';'.join(cm.output), cm.output)
-
-        inlet1.try_shutdown.assert_called()
-        outlet1.try_shutdown.assert_called()
-
-    @patch(fqname(Outlet), spec=Outlet)
-    @patch(fqname(Outlet), spec=Outlet)
-    @patch(fqname(Inlet), spec=Inlet)
-    def test_on_shutdown_outlet_exception_raise(self, inlet1, outlet1, outlet2):
-        outlet1.try_shutdown.side_effect = lambda: exec('raise(RuntimeError())')
-        link = Link([inlet1], [outlet1, outlet2], timedelta(seconds=1), name='test_on_shutdown')
-
-        self.assertRaises(RuntimeError, link.on_shutdown)
-
-        inlet1.try_shutdown.assert_called()
-        outlet1.try_shutdown.assert_called()
-        outlet2.try_shutdown.assert_not_called()
-
-    @patch(fqname(Outlet), spec=Outlet)
-    @patch(fqname(Outlet), spec=Outlet)
-    @patch(fqname(Inlet), spec=Inlet)
-    def test_on_shutdown_outlet_exception_catch(self, inlet1, outlet1, outlet2):
```

Comment:
Something went wrong here 🤔 These tests were added with the on_start/shutdown exception handling PR, somehow this PR removes them?

Reply:
hmmm that's not good. It must have been a conflict. Strange as I was trying to look out for these tests.

Reply:
Yupp, my IDE shows me it's this one:

https://github.com/Voyz/databay/pull/12/commits/98873e8e257b6692a81cc6792aae29e7f68f0afc

It's not visible in GH, but I think it was there. I'm gonna add these tests back on the dev directly, ok?